corehq/apps/builds/management/commands/commcare_build_importer.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import requests
+
+from cStringIO import StringIO
+from jenkinsapi.jenkins import Jenkins
+
+from corehq.apps.builds.models import CommCareBuild
+from django.core.management.base import BaseCommand, CommandError
+
+
+# supress requests library INFO logs
+logging.getLogger("requests").setLevel(logging.WARNING)
+
+
+class Command(BaseCommand):
+    """
+    This is an automated version of -
+    https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/builds#adding-commcare-j2me-builds-to-commcare-hq
+    """
+    help = "Interactively import a build from Jenkins build server."
+
+    def handle(self, *args, **options):
+        if not args:
+            self.interactive_import()
+        else:
+            raise CommandError(self.help)
+
+    def interactive_import(self):
+        print "Pinging Jenkins build server. Pelase wait..."
+        build_server = Jenkins("http://jenkins.dimagi.com")
+
+        # let user choose one of ['commcare-mobile-2.21', 'commcare-mobile-2.22']
+        jenkin_projects = build_server.keys()
+        selected_project_key = None
+        print "Jenkins has following projects. Please choose one"
+        print [p for p in jenkin_projects if 'commcare-mobile' in p]
+        while selected_project_key not in jenkin_projects:
+            selected_project_key = raw_input("")
+
+        print "Fetching build information for %s. Please wait..." % selected_project_key
+        selected_project = build_server[selected_project_key]
+        build_dict = selected_project.get_build_dict()
+        builds_by_version_number = self._extract_version_numbers(build_dict)
+
+        if not builds_by_version_number:
+            print "Jenkins doesn't have any builds with a VERSION set. Go do it yourself at "\
+                "http://jenkins.dimagi.com/job/%s", selected_project_key
+            return
+
+        selected_build_number = None
+        print "Jenkins has following builds for %s. Choose a build-number to import" % selected_project_key
+        print builds_by_version_number
+        while selected_build_number not in builds_by_version_number.keys():
+            selected_build_number = int(raw_input(""))
+
+        # download and add the build
+        version_number = builds_by_version_number.get(selected_build_number)
+        build = selected_project.get_build_metadata(selected_build_number)
+        artifacts = build.get_artifact_dict()
+        artifacts_url = artifacts['artifacts.zip'].url
+
+        print "Downloading and importing artifacts.zip. Please wait..."
+        zip_file = requests.get(artifacts_url)
+        self.add_build(StringIO(zip_file.content), version_number, selected_build_number)
```

Comment:
The first argument you're passing in here is a file-like object, not a path. Maybe rename `build_path` to `build_file` below?